src/content_mapblock.cpp

```diff
@@ -401,9 +405,15 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
 
 			// Neighbor liquid levels (key = relative position)
 			// Includes current node
-			std::map<v3s16, f32> neighbor_levels;
-			std::map<v3s16, content_t> neighbor_contents;
-			std::map<v3s16, u8> neighbor_flags;
+
+			struct NeighbourData
+			{
+				f32 level;
+				content_t content;
+				u8 flags;
+			};
+			NeighbourData neighbour_datas[27];
```

Comment:
Agreed, "datas" is ungrammatical.  "datum" wouldn't work either since it's a struct with three distinct datums that come together to form data, so it'd be misleading.  The best thing to do here would be to just rename the variable to "neighbour_data" IMHO.